lib/geo.js

```diff
@@ -9,80 +9,123 @@ var assert = require('assert');
 
 /*!
  * Get a near filter from a given where object. For connector use only.
+ * NB: This only supports one near parameter per where object.  eg, this:
+ * {where: {or: [{location: {near: "29,-90"}},{location: {near: "50,-72"}}]}}
+ * would throw an error.
  */
 
 exports.nearFilter = function nearFilter(where) {
-  var result = false;
-
-  if (where && typeof where === 'object') {
-    Object.keys(where).forEach(function(key) {
-      var ex = where[key];
-
-      if (ex && ex.near) {
-        result = {
-          near: ex.near,
-          maxDistance: ex.maxDistance,
-          unit: ex.unit,
-          key: key,
-        };
+  function nearSearch(clause, parentKeys) {
+    if (typeof clause !== 'object') {
+      return false;
+    }
+    if (!parentKeys) {
+      parentKeys = [];
+    }
+
+    Object.keys(clause).forEach(function(clauseKey) {
+      if (Array.isArray(clause[clauseKey])) {
+        clause[clauseKey].forEach(function(el, index) {
+          var ret = nearSearch(el, parentKeys.concat(clauseKey).concat(index));
+          if (ret) {
+            return ret;
+          }
+        });
+      } else {
+        if (clause[clauseKey].hasOwnProperty('near')) {
+          var result = clause[clauseKey];
+          nearResults.push({
+            near: result.near,
+            maxDistance: result.maxDistance,
+            unit: result.unit,
+            // If key is at root, define a single string, otherwise append it to the full path array
+            // key: parentKeys.length ? parentKeys.concat(clauseKey) : clauseKey,
+            key: clauseKey
```

Comment:
i rolled back the line from the mysql fix (see #993)

**L42** works for mongoDB but fails for the unit-tests (and would fail for mysql) as latter on for non mongoDB connectors the `filter()` method will be called and it expects a string, not an array for [`var loc = obj[key];`](https://github.com/strongloop/loopback-datasource-juggler/pull/1216/files#diff-b367a815d81f15531e0d0553ea8aaa5aR90).

so here's the situation: 
- how can we unit-test in the juggler the specific case where the juggler will call the geoNear operator under the hood (are there integration tests?)
- how to detect upstream in the `nearFilter()` method when to trigger one way (non mongoDB connector) or the other (mongoDB connecter) to build the `key` property ?